core/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenRevocationEndpointFilter.java

```diff
@@ -0,0 +1,149 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * A {@code Filter} for the OAuth 2.0 Token Revocation,
+ * which handles the processing of the OAuth 2.0 Token Revocation Request.
+ *
+ * @author Vivek Babu
+ * @see OAuth2AuthorizationService
+ * @see OAuth2Authorization
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7009#section-2">Section 2 Token Revocation</a>
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7009#section-2.1">Section 2.1 Revocation Request</a>
+ * @since 0.0.1
+ */
+public class OAuth2TokenRevocationEndpointFilter extends OncePerRequestFilter {
+
+	/**
+	 * The default endpoint {@code URI} for token revocation request.
+	 */
+	public static final String DEFAULT_TOKEN_REVOCATION_ENDPOINT_URI = "/oauth2/revoke";
+	private static final String TOKEN_TYPE_HINT = "token_type_hint";
+	private static final String TOKEN = "token";
+	private final AntPathRequestMatcher revocationEndpointMatcher;
+
+	private final Converter<HttpServletRequest, Authentication> tokenRevocationAuthenticationConverter =
+			new OAuth2TokenRevocationEndpointFilter.TokenRevocationAuthenticationConverter();
+	private final HttpMessageConverter<OAuth2Error> errorHttpResponseConverter =
+			new OAuth2ErrorHttpMessageConverter();
+	private final AuthenticationManager authenticationManager;
+
+	/**
+	 * Constructs an {@code OAuth2TokenRevocationEndpointFilter} using the provided parameters.
+	 *
+	 * @param authenticationManager the authentication manager
+	 */
+	public OAuth2TokenRevocationEndpointFilter(AuthenticationManager authenticationManager) {
+		this(authenticationManager, DEFAULT_TOKEN_REVOCATION_ENDPOINT_URI);
+	}
+
+	/**
+	 * Constructs an {@code OAuth2TokenRevocationEndpointFilter} using the provided parameters.
+	 *
+	 * @param authenticationManager the authentication manager
+	 * @param revocationEndpointUri the endpoint {@code URI} for revocation requests
+	 */
+	public OAuth2TokenRevocationEndpointFilter(AuthenticationManager authenticationManager,
+			String revocationEndpointUri) {
+		Assert.notNull(authenticationManager, "authenticationManager cannot be null");
+		Assert.hasText(revocationEndpointUri, "revocationEndpointUri cannot be empty");
+		this.authenticationManager = authenticationManager;
+		this.revocationEndpointMatcher = new AntPathRequestMatcher(
+				revocationEndpointUri, HttpMethod.POST.name());
+	}
+
+	@Override
+	protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+			throws ServletException, IOException {
+
+		if (!this.revocationEndpointMatcher.matches(request)) {
+			filterChain.doFilter(request, response);
+			return;
+		}
+
+		try {
+			Authentication tokenRevocationRequestAuthentication =
+					this.tokenRevocationAuthenticationConverter.convert(request);
+			this.authenticationManager.authenticate(tokenRevocationRequestAuthentication);
```

Comment:
We should explicitly set status to `200` - even though this will default. However, it will ensure the response is written and therefore will commit.